packages/idempotency/src/persistence/index.ts

```diff
@@ -1,3 +1,4 @@
 export * from './BasePersistenceLayer';
 export * from './BasePersistenceLayerInterface';
 export * from './IdempotencyRecord';
+export * from './DynamoDBPersistenceLayer';
```

Comment:
I left this out and exposed it via a separate import on purpose, and I feel strongly (but are willing to discuss) about keeping it that way.

At the moment we only have one persistence layer, so having it exported here is fine.

If (or actually when) we add more persistence layers (i.e. ElastiCache or others), this file will become:

```ts
// packages/idempotency/src/persistence/index.ts
export * from './DynamoDBPersistenceLayer';
export * from './OtherPersistenceLayer';
```

However **customers using one won't need the other**.

By having it like this, whenever customers do this: `import { DynamoDBPersistenceLayer } from '@aws-lambda-powertools/idempotency/persistence';` or this: `import { OtherPersistenceLayer } from '@aws-lambda-powertools/idempotency/persistence';` both are imported.

This a few undesirable effects:

- If any of the persistence layers have unique dependencies, compilers like `tsc` will require them to be installed in the project, even if you are not explicitly using the code (because you are importing it) - this has happened in #1068, resulted in #1225, and is something that we [want to change in v2](https://github.com/awslabs/aws-lambda-powertools-typescript/discussions/1269#discussioncomment-4891876) for Logger, Tracer, and Metrics.
- Some bundlers like `esbuild` cannot always tree shake the unused module, because it's always being imported
- If the initialization of the module is expensive (which is **not** the case with DynamoDB), then customers would be paying this price for both persistence layers.

By having them separate instead, customers have to explicitly opt-in and import whichever persistence layer they are actually using, i.e. `import { DynamoDBPersistenceLayer } from '@aws-lambda-powertools/idempotency/dynamodb';` or `import { OtherPersistenceLayer } from '@aws-lambda-powertools/idempotency/other';`. This way the imports are isolated from each other and: 1/ dependencies for each one can be installed selectively, 2/ tree-shaking and bundling can be done effectively.

This can be achieved with customizing the exports ([see here](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/idempotency/package.json#L44-L47)), and is also why in Parameters we don't have an `index.ts` file that exports all the different providers.

Reply:
Hmm I see, I was wondering about the inconsistency of the exports. I agree that we should be mindful of the exports. 

I added it to collect it for the typedoc output, since the entrypoint is `index.ts`. I will revert the exports to the previous state.